Extract obsolete operating systems from LDAP

Shad0wC0ntr0ller opened the pull request:
Extract enabled obsolete operating systems from LDAP

nxc ldap dcip -u user -p password -M obsolete

![image](https://github.com/Pennyw0rth/NetExec/assets/90877534/711cbb42-1c17-42af-8f2f-a2d9ac82da06)

zblurx commented:
Hey, I love this idea
However for parsing purpose, choosing arrow as the separator is not great (how can you awk the output ?)
What about `machinename (10.10.10.1) : OS name` ?
Or maybe a module option to set a delimiter or even a csv output file ?

Shad0wC0ntr0ller commented:
@zblurx you know, that’s a great point that has not been brought to my attention before. I will update the module in an hour or so. Thanks for the good call out.

Shad0wC0ntr0ller commented:
@zblurx I have made the adjustment. I also went ahead and added Server 2012 as that's less than a couple of weeks away. Finally i updated the output path from cme to nxc.

NeffIsBack commented:
Nice! Also it would probably be nice to give some feedback to the user about the file you are saving the output to. Something like sam and lsa dump do

Shad0wC0ntr0ller commented:
There is an info statement that shows where the file is saved to in the output already. 

![image](https://github.com/Pennyw0rth/NetExec/assets/90877534/387c13a4-959e-4922-83d9-9e952247cbd9)

Are you saying you want them to get added to a database like this ?

![image](https://github.com/Pennyw0rth/NetExec/assets/90877534/c85272e9-0bd8-4278-be9d-d778b2dcf20c)

NeffIsBack commented:
Oh my bad, missed that line. Just wanted to have the file path in some output but you already thought of that😄

zblurx commented on nxc/modules/obsolete.py:
```diff
@@ -0,0 +1,88 @@
+            hostname_parts = answers[0][0].split('.')
+            domain = ".".join(hostname_parts[1:])
+            
+            home = Path.home()
```
There is a more conventional way to get the logs folders in NXC -> `log_folder_path = os.path.join(os.path.expanduser("~/.nxc"), "logs")`

evildrummer commented:
Maybe adding the pwdlastset of the host? So you know it if the host is "kind a active"

Shad0wC0ntr0ller commented:
@evildrummer @zblurx Excellent ideas, I will look into this later today !

NeffIsBack commented:
@Shad0wC0ntr0ller any update on this?

Shad0wC0ntr0ller commented:
Sorry guys been busy with other things, ill get this handled by EOD today

NeffIsBack commented:
> Sorry guys been busy with other things, ill get this handled by EOD today

Awesome, but no pressure, justed wanted to check :)

Marshall-Hallenbeck commented:
@Shad0wC0ntr0ller did you want to get this up to date and working? We'd like to include it in the v1.2 release if you have time :)

Shad0wC0ntr0ller commented:
I will try to make the needed updates over the next couple of days. Thank you

Shad0wC0ntr0ller commented:
Sorry for the delay, I have updated the script it will now include the pwdlastset variable. let me know if there's any more adjustments you would like to see. thank you

NeffIsBack reviewed (approved):
Very nice addition! Thanks again for you work

NeffIsBack commented:
Working flawlessly:
![image](https://github.com/Pennyw0rth/NetExec/assets/61382599/feaac880-dd63-4a72-8c2b-ab85e8190d7a)

enj5oy commented:
@Shad0wC0ntr0ller hello man, can you explain when this info can help?

evildrummer commented:
@enj5oy it was a request from me. In PenTests you always check fir legacy or unsupported systems for the client.
It could also be a quick overview for easy targets
